packages/stylist/src/style_ext.rs

```diff
@@ -0,0 +1,64 @@
+use std::borrow::Borrow;
+use std::result::Result;
+use stylist_core::ast::Sheet;
+
+pub type Style = stylist_core::Style;
+
+/// Trait for types that can be parsed into css [`Sheets`].
+///
+/// [`Sheets`]: `stylist_core::ast::Sheet`
+pub trait TryParseCss {
+    /// Error type encountered during parsing
+    type Error;
+    /// Try to convert a self to a css [`Sheet`].
+    fn try_parse(self) -> Result<Sheet, Self::Error>;
+}
+
+impl<'a> TryParseCss for &'a str {
+    type Error = crate::error::Error;
+
+    fn try_parse(self) -> Result<Sheet, Self::Error> {
+        crate::parser::Parser::parse(self)
+    }
+}
+
+pub trait StyleExt: Sized {
+    /// Creates a new style with a custom class prefix from some parsable css.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use stylist::{Style, StyleExt};
+    ///
+    /// let style = Style::create("my-component", "background-color: red;")?;
+    /// # use stylist::TryParseCss;
+    /// # Ok::<(), <&str as TryParseCss>::Error>(())
+    /// ```
+    fn create<N: Borrow<str>, Css: TryParseCss>(classname: N, css: Css)
```

Comment:
What's the advantage of taking a custom trait over `AsRef<str>` for style sheet?

If it's a substitute for the `FromStr` trait on the original AST, maybe we can either move parser to stylist-core or just get away with it by using `Parser::parse` directly?

maybe prefix should be AsRef<str> as well?

Reply:
The point was to have different types of errors that are influenced by the parsing behavior. At first, I had one method and two impls, one for `&str`, another for `Style` that was Infallible and just cloned. I think in the course of the implementation I realized that most of the utility of Infallible isn't there yet, e.g. you can't easily with smth like [`Result::into_ok`](https://doc.rust-lang.org/std/result/enum.Result.html#method.into_ok) in stable rust, so I got rid of the second impl in favor of adding back the `new` constructors.

Tl;dr: Going back to `AsRef<str>` is fine by me.

Reply:
I would vote for having two different error types for parsing and mounting though